src/Library/demos/Color Dialog/main.js

```diff
@@ -0,0 +1,42 @@
+import Gtk from "gi://Gtk";
+import Gdk from "gi://Gdk";
+import Gio from "gi://Gio";
+
+Gio._promisify(Gtk.ColorDialog.prototype, "choose_rgba", "choose_rgba_finish");
+
+const color_dialog_button = workbench.builder.get_object("color_dialog_button");
+const custom_button = workbench.builder.get_object("custom_button");
+
+const color = new Gdk.RGBA();
+color.parse("white");
+
+const dialog_standard = new Gtk.ColorDialog({
+  title: "Select a color",
+  modal: true,
+  with_alpha: true,
+});
+
+color_dialog_button.dialog = dialog_standard;
+color_dialog_button.rgba = color;
+
+color_dialog_button.connect("notify::rgba", () => {
+  console.log(
+    `Color Dialog Button: The color selected is ${color_dialog_button.rgba.to_string()}`,
+  );
+});
+
+const dialog_custom = new Gtk.ColorDialog({
+  title: "Select a color",
+  modal: true,
+  with_alpha: false,
+});
+
+custom_button.connect("clicked", async () => {
```

Comment:
Just a piece of advice: be careful about using `async` functions as signal callbacks.

An `async` function will implicitly return a `Promise`, and if the signal handler takes a `boolean` return value the `Promise` will be coerced to `true`.